### PR TITLE
Capture usage metrics for scan and mirrors

### PR DESCRIPTION
**Pull request by munishchouhan**
This PR will add functionality to capture  usage metrics for wave scan and mirrors

### CODE REVIEW

File: src/main/groovy/io/seqera/wave/service/scan/ScanRequest.groovy [hunk 1]
```diff
@@ -78,6 +80,11 @@ class ScanRequest {
      */
     final Instant creationTime
 
+    /**
+     * Scan request platform identity
+     */
+    final PlatformId identity
```
Comment (pditommaso) [inline src/main/groovy/io/seqera/wave/service/scan/ScanRequest.groovy:86]: This method should be updated as well (and tests) 

[ScanRequest](https://github.com/seqeralabs/wave/blob/88fea7c69c9ff802612a8250ff43c868355d2690/src/main/groovy/io/seqera/wave/service/scan/ScanRequest.groovy#L82-L94)

**Comment by pditommaso** [comment]
Is it ready for you? have you made some tests ?

**Comment by munishchouhan** [comment]
> @munishchouhan Is it ready for you? have you made some tests ?

testing is not completed yet, finishing https://github.com/seqeralabs/wave/pull/701, then will test it

**Comment by pditommaso** [comment]
ok!

**Comment by munishchouhan** [comment]
scans tested locally:
Screenshot 2024-10-17 at 11 43 43

File: src/main/groovy/io/seqera/wave/service/scan/ScanRequest.groovy (hunk 1 above)
Comment (munishchouhan) [reply src/main/groovy/io/seqera/wave/service/scan/ScanRequest.groovy:86]: done

**Review by pditommaso — APPROVED** [review]
Excellent

**Pull request merged**